src/gsNurbs/gsKnotVector.hpp

```diff
@@ -991,13 +993,11 @@ template< typename T>
 void gsKnotVector<T>::supportIndex_into(const mult_t& i,
```

Comment:
Is this "only" a smarter implementation or does it change the results?

If the latter, we should change the documentation. Also, it is used in gsSmoothPatches in unsupported. Aren't we breaking it?

Reply:
For open knot vectors no change.
For vectors with ghost knots, they get negative indices, so we get different results.
The doc is OK, actually to keep the function doing the same thing I extended it to non-open knot vectors (which have knots with negative unique indices).

gsSmoothPatches in unsupported is assuming and using open knot vectors, so no change in this respect.